Realm/RLMResults.h

```diff
@@ -302,9 +302,16 @@ __attribute__((warn_unused_result));
 - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMResults<RLMObjectType> *_Nullable results,
                                                          RLMCollectionChange *_Nullable change,
                                                          NSError *_Nullable error))block
+                                      keyPaths:(nullable NSArray<NSString *> *)keyPaths
```

Comment:
Could `KeyPathArray` be used here?

Reply:
Like this `keyPaths:(nullable KeyPathArray)keyPaths`?
or this `keyPathArray:(nullable NSArray<NSString *> *)keyPathArray`?

Reply:
I think using a typedef here would just make it harder to use. The argument label already conveys the same information, and using KeyPathArray means that someone reading this then has to go look up what a KeyPathArray is.

Reply:
How could `KeyPathArray` be used instead of (NSArray<NSString *> *) for the parameter type? The developer will be passing in strings.

Reply:
Sorry, my bad! You're absolutely right. This is the actual SDK layer that the consumer uses so it makes a lot of sense there. :) I got lost in all the layers we have for a moment ..